src/main/commandServer/httpCommandServer.ts

```diff
@@ -176,6 +181,88 @@ export class HttpCommandServer {
     return undefined;
   }
 
+  protected diagnosticCategories(request: http.IncomingMessage, response: http.ServerResponse, context: commandContext): Promise<void> {
+    const settings = this.commandWorker.getDiagnosticCategories(context);
+
+    if (settings) {
+      console.debug('diagnosticCategories: succeeded 200');
+      response.writeHead(200, { 'Content-Type': 'text/plain' });
+      response.write(settings);
+    } else {
+      console.debug('diagnosticCategories: failed 404');
+      response.writeHead(404, { 'Content-Type': 'text/plain' });
+      response.write('No diagnostic categories found');
+    }
+
+    return Promise.resolve();
+  }
+
+  protected diagnosticIDsForCategory(request: http.IncomingMessage, response: http.ServerResponse, context: commandContext): Promise<void> {
+    const url = new URL(`http://${ request.url }`);
+    const searchParams = url.searchParams;
+    const category = searchParams.get('category');
+
+    if (!category) {
+      console.debug('diagnostic_ids: failed 400');
+      response.writeHead(400, { 'Content-Type': 'text/plain' });
+      response.write('diagnostic_ids: no  category specified');
```

Comment:
```suggestion
      response.write('diagnostic_ids: no category specified');
```
Should this be returned in JSON?

Reply:
Extra space removed.

You get the string `diagnostic_ids: no category specified` written to stderr and a JSON payload of `{ "message": "400 Bad Request" }' written to stdout.

Reply:
I'm confused: it says `response.write(<non-JSON string>);`, so the other end of the HTTP connection sees plain text?

(Leaving this one resolved, but please still update either this or the OpenAPI spec — the spec claims JSON, but this implements `text/plain`)